lib/ddtrace/contrib/extensions.rb

```diff
@@ -60,11 +73,23 @@ def configure(target = configuration, opts = {})
         module Settings
           InvalidIntegrationError = Class.new(StandardError)
 
-          def self.included(base)
-            # Add the additional options to the global configuration settings
-            base.instance_eval do
-              option :registry, default: Registry.new
-            end
+          # The registry only holds declarative constant values and cannot be modified.
+          # This option is a no-op and will be removed in the future.
+          #
+          # @deprecated Use `Datadog::Contrib::REGISTRY` instead
+          def registry
+            Datadog.logger.warn('Deprecated access to `Datadog.configuration.registry`, use `Datadog.registry` instead.' \
+                                '`Datadog.configuration.registry` will be removed in a future version.')
+            Contrib::REGISTRY
+          end
+
+          # The registry only holds declarative constant values and cannot be modified.
+          # This option is a no-op and will be removed in the future.
+          #
+          # @deprecated The registry is now a global constant, and can be overwritten.
```

Comment:
Well, yes, but do we really want to suggest that users do that? I think we should omit this: if users want to do weird things, well, this is Ruby, home of the monkey patch, but I'd stop short of actually suggesting it as a replacement for the removed setter.

Reply:
`can` -> `can't` a typo in my part.

Reply:
Ahhh wow the meaning changed quite a lot from that missing negative :)